src/psyclone/domain/gocean/transformations/gocean_const_loop_bounds_trans.py

```diff
@@ -104,25 +108,81 @@ def validate(self, node, options=None):
+        :type options: dict of str:values or None
 
         :raises TransformationError: if the supplied node is not a \
-                                     GOInvokeSchedule.
+            GOInvokeSchedule.
+        :raises TransformationError: if the supplied schedule has loops with \
+            index offsets
 
         '''
         if not isinstance(node, GOInvokeSchedule):
-            raise TransformationError("Error in GOConstLoopBoundsTrans: "
-                                      "node is not a GOInvokeSchedule")
+            raise TransformationError(
+                "GOConstLoopBoundsTrans can only be applied to "
+                "'GOInvokeSchedule' but found '{0}'."
+                "".format(type(node).__name__))
+
+        for loop in node.walk(GOLoop):
+            if loop.loop_type not in ["inner", "outer"]:
+                raise TransformationError(
+                    "GOConstLoopBoundsTrans can not transform a loop with "
+                    "loop_type '{0}', only 'inner' or 'outer' loop_type values"
+                    " are expected.".format(loop.loop_type))
+
+            if loop.index_offset not in loop.bounds_lookup:
+                raise TransformationError(
+                    "GOConstLoopBoundsTrans can not transform a loop with "
+                    "index_offset '{0}' because it is not in the bounds lookup"
+                    " table, the available index_offset values are {1}."
+                    "".format(loop.index_offset,
+                              list(loop.bounds_lookup.keys())))
+
+            table = loop.bounds_lookup[loop.index_offset]
+            if loop.field_space not in table:
+                raise TransformationError(
+                    "GOConstLoopBoundsTrans can not transform a loop with "
+                    "field_space '{0}' because it is not in the bounds lookup"
+                    " table, the available field_space values are {1}."
+                    "".format(loop.field_space, list(table.keys())))
+
+            table = table[loop.field_space]
+            if loop.iteration_space not in table:
+                raise TransformationError(
+                    "GOConstLoopBoundsTrans can not transform a loop with "
+                    "iteration_space '{0}' because it is not in the bounds "
+                    "lookup table, the available iteration_space values are "
+                    "{1}.".format(loop.iteration_space, list(table.keys())))
+
+            table = table[loop.iteration_space]
+            if loop.loop_type not in table:
+                raise TransformationError(
+                    "GOConstLoopBoundsTrans can not transform a loop with "
+                    "loop_type '{0}' because it is not in the bounds "
+                    "lookup table, the available loop_type values are "
+                    "{1}.".format(loop.loop_type, list(table.keys())))
+
+        # Make sure the Invoke has at least one field argument
+        found = False
```

Comment:
Please change this to use:

    for arg....:
         break
    else:
        raise TransformationError(...)

and then you can remove the `found` variable.

Reply:
Done